### PR TITLE
Show search filters by default

### LINKED ISSUE #22228: Showcase: Show search filters by default
#### Steps to reproduce
1. Go to Theme Showcase.
2. Click into the search filed.
3. The filters appear.

#### What I expected
I didn't expect the filters to appear as a new user. The filters are hidden from users who are not familiar with this action.

#### Screenshot / Video
show-filters

#### Context / Source
While this isn't a bug, it is a mystery meat sort of feature. The filters help guide user's queries, but only if they're visible. By showing them as default, it could help users understand how to search more specifically.

In addition to this, when the user has committed to entering text in the search field, they are likely not going to be clicking on filters at that point in their journey. So by showing them when the user is engaged in something else, we're distracting the user from their desired action.

### PR DESCRIPTION
**Pull request by mapk**
Show search filters by default. Fixes #22228

### CODE REVIEW

File: client/my-sites/themes/themes-magic-search-card/style.scss [hunk 1]
```diff
@@ -144,14 +144,9 @@
...
 
-	.themes-magic-search-card__welcome {
-		display: none;
-	}
-
 	&.has-keyed-suggestions {
 		.themes-magic-search-card__welcome,
 		.suggestions {
-			display: block;
 			margin-top: 5px;
```
Comment (mattsherman) [inline client/my-sites/themes/themes-magic-search-card/style.scss]: The slight jump from the `margin-top` when the search field has focus feels off. I would suggest removing this `margin-top` completely.

**Comment by mapk** [comment]
Looks like I've found another issue here though... if you don't have focus on the input field, and click one of the filters, the filter appears in the search field, but the list of filters doesn't show up. 

filter-issue

**Comment by mapk** [comment]
With this last commit came another result.

**Current**
After selecting a first-level filter (ie. Feature) and then clicking out of the input, the dropdown of filters disappears, but the filter name is still in the search box (ie. Filter:).

original-blur

**Commit change**
After selecting a first-level filter and then clicking out of input, the filter remains in the input field (as it did before), and so does the dropdown list.

filter-fix

This forces the user to click the 'X' in the input field to remove the filter. While writing this, I don't think this is good for mobile. I'm revisiting again to get the dropdown to disappear when clicking outside the input field. Expect another commit soon.

**Comment by mapk** [comment]
So, I've been able to keep the 'close' icon showing even on mobile, but the dropdown still doesn't disappear when clicking outside of the input. 

close-mobile

**Comment by mattsherman** [comment]
I changed the status back to "In Progress". When it is ready for another look change it back to "Needs Review" and I'll take another look!

**Comment by mattsherman** [comment]
@mapk Just let me know when this is ready for me to take another look at it and I'll review it. I want to make sure I don't miss/overlook when this is ready, as I started reviewing it. :-)

**Comment by lancewillett** [comment]
I'm happy to review, test, "QA", and help get this in, too.

**Comment by mapk** [comment]
@lancewillett that would be super helpful!

**Comment by kjellr** [comment]
> So, I've been able to keep the 'close' icon showing even on mobile

I was just testing, and wanted to note that I'd expect to see an X in here for the whole time the field is focused. Tapping outside the field will be like a good hidden feature, but there's still no obvious "escape" to the field when it's open. When tapping outside to close it, I frequently ended up triggering other elements on the page.

screen shot 2018-03-06 at 2 13 46 pm

**Comment by kjellr** [comment]
One other thing to note: When you select a list item, it automatically shifts focus to the search field. On mobile, this makes the keyboard appear. When that happens, you have to manually dismiss the keyboard before you can view the themes you just filtered to see:

keyboard

Perhaps we can remove that auto-focus on the text field in this case?

**Comment by mapk** [comment]
@kjellr, I think there are so many little nuances with this whole hierarchy of filters and placement of everything, that maybe we just merge this now (because I think your point existed prior to this PR anyway) and move on to fixing the filters and auto-focus in another PR?

**Comment by mapk** [comment]
@mattsherman All ready for review now! :)

**Comment by kjellr** [comment]
> I think your point existed prior to this PR anyway) and move on to fixing the filters and auto-focus in another PR?

Sounds good to me, @mapk. I opened a new issue here: #23125

**Comment by mapk** [comment]
@mattsherman I don't think I understand what the failed check is. Can you help? Should I merge anyway?

**Comment by mattsherman** [comment]
@mapk I believe the `ci/wp-desktop` failed check is because this PR was branched from master a while ago, and thus this branch is still using an older version of node. I think that if you rebase this PR against master that things will be fixed up.

**Pull request merged**